nexus/src/db/model.rs

```diff
@@ -2058,6 +2139,13 @@ impl ConsoleSession {
     }
 }
 
+// Return this instead, by performing an additional SELECT to get silo id
```

Comment:
I partially agree. I moved `ConsoleSessionWithSiloId` out to `authn` in d8aa2431 but I don't want Nexus returning a ConsoleSession without a Silo ID - I want to avoid the case where that can be passed around and consumed, and keep the version without the silo user id only in the DB.

Reply:
Would it be bizarre to store the `silo_id` on the session, and then `ConsoleSessionWithSiloId` just _is_ `ConsoleSession`?

Reply:
I now see why you originally had `ConsoleSessionWithSiloId` in models. It does feel like a model-level construct because it's fundamental to a session that it be attached to a silo. The link to a silo just happens to go through the user ID.

The thing is, `ConsoleSessionWithSiloId` is the true session model here because you can't _do_ anything with plain `ConsoleSession` except create one. I'm almost inclined to rename `ConsoleSession` to `ConsoleSessionDB` and `ConsoleSessionWithSiloId` to `ConsoleSession`, at least I would if it didn't break the convention we have that lets you assume if you see a model that there's a corresponding table with that exact name. The above suggestion of storing `silo_id` in the sessions table isn't very good, it's just denormalizing the data in order to only have one console session struct that complies with the convention.

Reply:
I think it's useful that there be a low-level layer that represents exactly what's in the database, which is what `db::model` is today (though it doesn't have to be).  I think we're both saying there's room for another layer atop that with more usable/useful abstractions.  We could make that an explicit layer ("data"? ugh the naming is hard) but that feels too bureaucratic to me.  I think it makes sense to put such abstractions in the subsystems they're associated with, which is why I suggested `authn` here.  But maybe it should just be `authn::ConsoleSession`?  The idea is a reader would understand: `db::model::X` is a low-level database interface but `some_subsystem::X` is probably what I want to be using, if it exists.